shell/platform/windows/compositor_opengl_unittests.cc

```diff
@@ -0,0 +1,192 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include <memory>
+#include <vector>
+
+#include "flutter/impeller/renderer/backend/gles/gles.h"
+#include "flutter/shell/platform/windows/angle_surface_manager.h"
+#include "flutter/shell/platform/windows/compositor_opengl.h"
+#include "flutter/shell/platform/windows/flutter_windows_view.h"
+#include "flutter/shell/platform/windows/testing/engine_modifier.h"
+#include "flutter/shell/platform/windows/testing/flutter_windows_engine_builder.h"
+#include "flutter/shell/platform/windows/testing/mock_window_binding_handler.h"
+#include "flutter/shell/platform/windows/testing/windows_test.h"
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+
+namespace flutter {
+namespace testing {
+
+namespace {
+using ::testing::Return;
+
+const unsigned char* MockGetString(GLenum name) {
+  switch (name) {
+    case GL_VERSION:
+    case GL_SHADING_LANGUAGE_VERSION:
+      return (unsigned char*)"3.0";
+    default:
+      return (unsigned char*)"";
+  }
+}
+
+void MockGetIntegerv(GLenum name, int* value) {
+  *value = 0;
+}
+
+GLenum MockGetError() {
+  return GL_NO_ERROR;
+}
+
+void DoNothing() {}
+
+const impeller::ProcTableGLES::Resolver kMockResolver = [](const char* name) {
```

Comment:
Guessing this doesn't support `std::string_view` which would get you nice operator== semantics?

Reply:
Right, this mirrors `eglResolveFunction`. I updated this to allocate convert the `name` into a `std::string` to get those nice `operator==` semantics :)

Reply:
You could even use std::string_view and get the operator== emantics roughly for free without an additional allocation/copy.